sdk/typescript/src/signers/signer.ts

```diff
@@ -31,4 +36,61 @@ export abstract class Signer {
    * Returns the signature for the data and the public key of the signer
    */
   abstract signData(data: Base64DataBuffer): Promise<SignaturePubkeyPair>;
+
+  // Returns a new instance of the Signer, connected to provider.
+  // This MAY throw if changing providers is not supported.
+  abstract connect(provider: Provider): Signer;
+
+  ///////////////////
+  // Sub-classes MAY override these
+
+  /**
+   * Sign a transaction and submit to the Gateway for execution
+   *
+   * @param txBytes BCS serialised TransactionData bytes
+   */
+  async signAndExecuteTransaction(
+    txBytes: Base64DataBuffer
+  ): Promise<TransactionResponse> {
+    this._checkProvider('signAndExecuteTransaction');
+    const sig = await this.signData(txBytes);
+    return await this.provider!.executeTransaction({
+      txBytes: txBytes.toString(),
+      signature: sig.signature.toString(),
+      pubKey: sig.pubKey.toString(),
+    });
+  }
+
+  /**
+   * Serialize and Sign a `Transfer` transaction and submit to the Gateway for execution
+   */
+  async transfer(
+    transaction: TransferTransaction
+  ): Promise<TransactionResponse> {
+    this._checkProviderAndSerializer('transfer');
+    const txBytes = await this.serializer!.new_transfer(transaction);
+    return await this.signAndExecuteTransaction(txBytes);
+  }
+
+  ///////////////////
+  // Sub-classes SHOULD leave these alone
+
+  _checkProviderAndSerializer(operation?: string): void {
```

Comment:
can we also eliminate all `_` prefixes here ?  an explicit typescript `private` modifier would communicate the intent more clearly